### DISCUSSION TITLE
do i need tmux installed before split screen works on linux?

### DISCUSSION THREAD

**Discussion by hillonyechekwa** [wezterm/wezterm#403] [General]
Hi, I've been finding it difficult to use the split screen feature so I looked through the issue on the repo for anything about the split screen feature but i keep seeing tmux mentioned.  Do i need to have tmux installed first,, cos every time i type in the ctlr+shift+alt+" or % command nothing happens.

**Comment by aorith** [comment]
Hey!

Try to remap it in your config, see https://wezfurlong.org/wezterm/config/keys.html#configuring-key-assignments

I had to remap "increase font size" binding since the default didn't work on my keyboard layout.

Also make sure you're running a nightly version since I think splits were implemented recently, you don't need tmux.

**Reply by hillonyechekwa** [reply]
tried doing this but it kept giving me this thing `expected a '{' after SUPER`

**Reply by hillonyechekwa** [reply]
![wezconfig](https://user-images.githubusercontent.com/46321192/103297138-f882bf00-49f7-11eb-8998-d76b35f36044.png)

**Reply by hillonyechekwa** [reply]
maybe i need another eyes to look at it

**Reply by wez** [reply]
The quoting isn't quite right you want something like this:

```lua
local wezterm = require 'wezterm';
return {
  keys = {
    key="\"", mods="SUPER", action=wezterm.action{SplitHorizontal={domain="CurrentPaneDomain"}},
    key="|", mods="SUPER", action=wezterm.action{SplitVertical={domain="CurrentPaneDomain"}},
  },
}
```

**Reply by wez** [reply]
You can set `debug_key_events = true` in your config to have wezterm log which keys are being pressed to its stderr stream (so you'll want to run wezterm from a terminal to see that logging); that might reveal what is happening with the modifiers.  Depending on your window manager, you may find that the window manager takes precedence for certain key combinations

**Reply by hillonyechekwa** [reply]
ok thanks. I'll try this now

**Reply by hillonyechekwa** [reply]
so I've done that but now a window pops up with this error message

`Error converting lua value returned by script /home/thegridsmith/.wezterm.lua
to Config struct: while processing a struct of type `Config` with value:
{
    "color_scheme": "nord",
    "cursor_blink_rate": 900,
    "debug_key_events": true,
    "default_cursor_stye": "BlinkingBlock",
    "dpi": 98,
    "enable_scroll_bar": true,
    "enable_tab_bar": true,
    "font": {
        "font": [
            {
                "bold": false,
                "family": "Fira Code Retina",
                "is_fallback": false,
                "italic": false,
            },
        ],
    },
    "font_size": 12,
    "keys": {
        "action": {
            "SplitVertical": {
                "domain": "CurrentPaneDomain",
                "set_environment_variables": {},
            },
        },
        "key": "|",
        "mods": "SUPER",
    },
    "line_height": 1.2,
    "scrollback_lines": 3500,
    "term": "xterm-256color",
    "window_background_opacity": 0.8,
    "window_close_confirmation": "AlwaysPrompt",
}
invalid type: map, expected a sequence
`

**Reply by wez** [reply]
please share exactly what is in your wezterm.lua config file; that error means that something isn't quite the right shape in the config, but because of the way that the data gets mapped from lua to rust, it isn't totally sure where the issue is :-/

**Reply by wez** [reply]
it's possible that the empty `set_environment_variables` list is the trigger for this; either put some environment variables inside it, or comment it out

**Reply by hillonyechekwa** [reply]
here it is
![wez -lua-config](https://user-images.githubusercontent.com/46321192/103302706-8a90c480-4a04-11eb-8976-c36894fdf7af.png)

**Reply by wez** [reply]
ah, I think I see it: the `keys` list isn't a list; that's a transcription error in my example above; note the difference here: each key is contained in `{ }` so that it is a separate object:

```lua
local wezterm = require 'wezterm';
return {
  keys = {
    {key="\"", mods="SUPER", action=wezterm.action{SplitHorizontal={domain="CurrentPaneDomain"}}},
    {key="|", mods="SUPER", action=wezterm.action{SplitVertical={domain="CurrentPaneDomain"}}},
  },
}
```

**Reply by wez** [reply]
(Please avoid pasting screenshots of configs: it makes it more accessible, readable and easier for others to copy and amend if you share the text itself!)

**Reply by hillonyechekwa** [reply]
alright. Thanks alot

**Comment by wez** [comment]
You don't need tmux to use splits in wezterm.  The most recent release has splits.  However, depending on the OS you're using, you may want to use a recent nightly build as a few keyboard shortcut issues have been resolved and that might explain the issue you described!

If that doesn't work out, please let us know the wezterm version and platform that you're running!

**Reply by hillonyechekwa** [reply]
I'm using the Debian 10 version, on pop OS (Linux)

**Reply by wez** [reply]
What does `wezterm --version` output?

**Reply by hillonyechekwa** [reply]
![wezconfig2](https://user-images.githubusercontent.com/46321192/103297011-b0639c80-49f7-11eb-9fd5-dce9220ac045.png)

**Reply by hillonyechekwa** [reply]
I'm running the nightly version
